packages/plugin-bundle-webpack/plugin.js

```diff
@@ -0,0 +1,139 @@
+const jsdom = require("jsdom");
+const CopyPlugin = require("copy-webpack-plugin");
+const { JSDOM } = jsdom;
+
+async function compilePromise(webpackConfig) {
+  const compiler = webpack(webpackConfig);
+  return new Promise((resolve, reject) => {
+    compiler.run((err, stats) => {
+      if (stats.hasErrors()) {
+        reject(err);
+      }
+      resolve(stats);
+    });
+  });
+}
+
+module.exports = function plugin(config, args) {
+  return {
+    async bundle({ srcDirectory, destDirectory, log, jsFilePaths }) {
+      let homepage = config.homepage || "";
+      let fallback = config.devOptions?.fallback || "index.html";
+
+      let dom = new JSDOM(
+        fs.readFileSync(path.join(srcDirectory, config.devOptions.fallback))
+      );
+
+      //Find the first local script, assume it's the entrypoint
+      let scriptEl;
+      for (const el of Array.from(
+        dom.window.document.querySelectorAll("script")
+      )) {
+        let src = el.src.trim();
+        if (
+          el.type.trim().toLowerCase() === "module" &&
+          !src.trim().match(/[a-zA-Z+]+:\/\//g)
+        ) {
+          scriptEl = el;
+          break;
+        }
+      }
+      if (!scriptEl) {
+        throw new Error("Can't bundle without script tag in html");
+      }
+      let entryPoint = scriptEl.src;
+
+      //Compile files using webpack
+      let webpackConfig = {
+        resolve: {
+          alias: {
+            "/web_modules": path.join(srcDirectory, "web_modules/"),
+          },
+        },
+        module: {
+          rules: [
+            {
+              test: /\.jsx?$/,
+              exclude: ["/node_modules/"],
+              use: {
+                loader: "babel-loader",
+              },
+            },
+            {
+              test: /\.s?css$/,
+              use: [
+                {
+                  loader: MiniCssExtractPlugin.loader,
+                },
+                {
+                  loader: "css-loader",
+                },
+              ],
+            },
+            {
+              test: /\.(svg|png|jpe?g|gif)$/, //We can expand this list of importable files
+              use: [
+                {
+                  loader: "file-loader",
+                  options: {
+                    name: "assets/[name]-[hash].[ext]",
+                  },
+                },
+              ],
+            },
+          ],
+        },
+        mode: "production",
+        plugins: [
+          //Extract a css file from imported css files
+          new MiniCssExtractPlugin({
+            filename: "css/style-[hash].css",
+          }),
+          //Copy other files to the destination, excluding ones that are no longer useful
+          new CopyPlugin({
+            patterns: [
+              {
+                from: srcDirectory,
+                to: destDirectory,
+                globOptions: {
+                  ignore: [`/${fallback}`, "**/_dist_/**", "**/web_modules/**"],
```

Comment:
`_dist_` is CSA specific, and not general so we should avoid baking it into the plugin. But, I think this should be safe to remove anyway.

Reply:
But if we remove _dist_ from ignore, the _dist_ folder is going to get copied over from .build to build, isn't it?

Reply:
hmm maybe, lets keep it in for now then